### ISSUE TITLE
flows visualisation with arcs or flowmaps

### ISSUE THREAD

**Issue by e-kotov** [walkerke/mapgl#37]
Hello @walkerke , I wonder if it is possible to add support for flow visualisation with both or either of:
- arcs https://svelte-maplibre.vercel.app/examples/deckgl-arcs , implemented in https://symbolixau.github.io/mapdeck/reference/add_arc.html 

![](https://github.com/SymbolixAU/mapdeck/raw/master/vignettes/img/readme_arcs_small.gif)

- flowmaps https://github.com/FlowmapBlue/FlowmapBlue ; https://github.com/FlowmapBlue/flowmapblue.R

![](https://github.com/FlowmapBlue/flowmapblue.R/blob/master/vignettes/webmedia/demo.gif?raw=true)

**Comment by bt-maps** [comment]
See also https://github.com/JohMast/flowmapper

**Comment by e-kotov** [comment]
> See also https://github.com/JohMast/flowmapper

yes, see also, but these are static. I do not suggest implementing the algorithm used in `{flowmapper}`.

I was thinking since the underlying tech of `{mapgl}`, `{deckgl}`, `{mapdeck}`, `{flowmapblue}` are essentially the same (correct me if i'm wrong, but it's all based on deck.gl one way or another), adding arcs or even better, flowmaps, to `{mapgl}` (which to me seems like the most ergonomic solution with many nice features) may be relatively easy.

**Comment by bt-maps** [comment]
Agreed that deck.gl/flowmap.gl is preferred. Deck.gl layer sources would be
a great feature. Py-maplibregl has this functionality for shiny for python.
https://eodagmbh.github.io/py-maplibregl/

flowmapper algorithm is from flowmapblue, so some additional work would be
needed to visualize with mapgl using sf objects

**Comment by walkerke** [comment]
This would be interesting to consider.  Those types of maps are available in deck.gl, which is not part of Mapbox GL JS or MapLibre GL JS by default.  So we'd need to build out a deck.gl interface to accomplish that (which is what py-maplibregl has done).  I'm not sure what the development lift for that would be.  

In the meantime, the rdeck package (https://github.com/qfes/rdeck) is also a really nice interface to deck.gl (though unfortunately not on CRAN).  You might consider taking a look at it for flow mapping.

**Comment by walkerke** [comment]
Arc visualization support may be coming soon to Mapbox, which means we'll be able to support it here.  See: https://github.com/mapbox/mapbox-gl-js/issues/3993#issuecomment-2533717645

**Comment by e-kotov** [comment]
Non-interactive flows are now mappable with `{mapgl}`, thanks to the latest update in [`{flowmapper}`](https://github.com/JohMast/flowmapper) by @JohMast! 

![Image](https://github.com/user-attachments/assets/ecb74dd2-7f0b-46ec-8e7a-7be4b9d1673a)

Code and discussion over here:
https://github.com/JohMast/flowmapper/issues/7#issuecomment-2809270589

**Comment by walkerke** [comment]
This is excellent @e-kotov!

**Comment by walkerke** [comment]
closing this as @e-kotov your implementation is looking quite impressive externally!

**Comment by walkerke** [comment]
ah ok- thought this was shaping up as a plugin rather than a PR.  reopening
